File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.py
import math

def solution(r1, r2):
    answer = 0
    
    # 작은 원 위의 4개 점 + 큰 원 위의 4개 점 + 그 사이 공간의 점 개수 * 4
    
    # 큰 원에서 y좌표값 - 작은 원에서 y좌표값
    # x**2 + y**2 = r**2 -> Math.sqrt(r**2 - x**2)
    for i in range(1, r2+1):
        e = math.floor(math.sqrt(r2**2 - i**2)) # 큰 원 내부
        if r1 > i:
            s = math.ceil(math.sqrt(r1**2 - i**2)) # 작은 원 외부
        else:
            s = 0
        answer += e - s + 1
    
    answer = answer*4 
    
    return answer

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.java
class Solution {
    public long solution(int r1, int r2) {
        long answer = 0;
        
        for (int i=0; i<=r2; i++) {
            double s = 0;
            
            if (i<r1) {
                s = Math.ceil(Math.sqrt(Math.pow(r1,2)-Math.pow(i,2)));
            }
            double e = Math.floor(Math.sqrt(Math.pow(r2,2)-Math.pow(i,2)));
            
            // 큰원에서의 y좌표 - 작은원에서의 y좌표
			answer += e - s + 1;
        }
        
        long minus = (r2 - r1 + 1) * 4;
        return answer * 4 - minus;
    }
}